src/dataframe/dataframe.jl

```diff
@@ -818,15 +818,12 @@ function select!(df::DataFrame, inds::AbstractVector{Int})
     if !allunique(inds)
         throw(ArgumentError("indices must not contain duplicates"))
     end
-
-    targetnames = _names(df)[inds]
-    for i in setdiff(ncol(df):-1:1, inds)
-        splice!(_columns(df), i)
-        delete!(index(df), i)
+    copy!(_names(df), _names(df)[inds])
+    copy!(_columns(df), _columns(df)[inds])
```

Comment:
```suggestion
    copy!(_columns(df), view(_columns(df), inds))
```

Reply:
This code will not work correctly because of aliasing. We need to copy:
```
julia> x = [1,2,3,4,5]
5-element Array{Int64,1}:
 1
 2
 3
 4
 5

julia> y = view(x,[2,1])
2-element view(::Array{Int64,1}, [2, 1]) with eltype Int64:
 2
 1

julia> copy!(x, y)
2-element Array{Int64,1}:
 2
 2

julia> x
2-element Array{Int64,1}:
 2
 2
```
fortunately here it is not expensive in comparison to other operations.

Reply:
Ah right. Copying is fine.

Ideally we could simply write `deleteat!(x, Not(inds))` or maybe `filter!(x, inds), but that's not a big deal.